src/qibolab/designs/basic.py

```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass, field
+
+from qibolab.instruments.abstract import AbstractInstrument
+
+
+@dataclass
+class BasicInstrumentDesign:
+    """Instrument design that uses a single controller.
+
+    Attributes:
+        controller (:class:`qibolab.instruments.abstract.AbstractInstrument`): Instrument used for sending pulses and retrieving feedback.
+        is_connected (bool): Boolean that shows whether instruments are connected.
+    """
+
+    controller: AbstractInstrument
+    channels: dict = field(default_factory=dict)
+    is_connected: bool = field(default=False, init=False)
```

Comment:
I believe this attribute could be private, `_is_connected`, no other classes are supposed to use this directly, right?

Reply:
Done.